src/domain.rs

```diff
@@ -442,17 +457,17 @@ fn fft_composition() {
             }
 
             let mut domain = EvaluationDomain::from_coeffs(v.clone()).unwrap();
-            domain.ifft(&worker);
-            domain.fft(&worker);
+            domain.ifft(&worker, &mut None);
+            domain.fft(&worker, &mut None);
             assert!(v == domain.coeffs);
-            domain.fft(&worker);
-            domain.ifft(&worker);
+            domain.fft(&worker, &mut None);
+            domain.ifft(&worker, &mut None);
             assert!(v == domain.coeffs);
-            domain.icoset_fft(&worker);
-            domain.coset_fft(&worker);
+            domain.icoset_fft(&worker, &mut None);
+            domain.coset_fft(&worker, &mut None);
             assert!(v == domain.coeffs);
-            domain.coset_fft(&worker);
-            domain.icoset_fft(&worker);
+            domain.coset_fft(&worker, &mut None);
+            domain.icoset_fft(&worker, &mut None);
```

Comment:
maybe change the type to `Option<&mut X>`, that way you can just pass a `None` here

Reply:
Hey @dignifiedquire, thank you for the comments!
I'm not sure how to do that, `Option<FFTKernel>` is shared among all of the fft calls in `prover.rs`, so passing it as a non-reference results in a move that invalidates it in the next fft calls. I'm not a professional Rust programmer so there is probably a workaround for this.